app/middleware/authentication.middleware.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+var userController = require('../controllers/user.controller');
+var config = require('../config/config');
+var q = require('q');
+
+// middleware
+exports.authentication = function(request, response, next) {   
+    var email = request.headers['x-forwarded-email']
+    var name = request.headers['x-forwarded-user']
+    if (!email) {
+        return response.status(401).json({ message: 'You need to login first!' });
+    }
+
+    userController.getUserByEmail(email)
+        .then(function(user) {
+            if (!user) {
+                var user = userController.getUserTemplate();
+                user.name = name;
+                user.email = email;
+                userController.createNewUser(user)
```

Comment:
Could `createNewUser` take `name` and `email` as parameters, and call `getUserTemplate` internally? Would that result in nicer looking code?